connector-packager/connector_packager/xsd_validator.py

```diff
@@ -176,3 +180,12 @@ def validate_file_specific_rules(file_to_test: ConnectorFile, path_to_file: Path
                     return False
 
     return True
+
+# Check if connector file content contains warnings needs to notify connector developer
+def check_file_content(file_to_test: ConnectorFile, path_to_file: Path):
+
+    if file_to_test.file_type == 'dialect':
```

Comment:
Would this logic fit within `validate_file_specific_rules`?

Reply:
I tried to add it there when I started this work, decided not to because 1. `validate_file_specific_rules` is validating file, if not valid, will return false, but for this change, we want a warning, if find defaultSQLDialect as base, we still return true
2. `validate_file_specific_rules` is called twice in packaging process, one is at [gererate_file_list](https://github.com/tableau/connector-plugin-sdk/blob/master/connector-packager/connector_packager/package.py#L89), two is at [validate_all_xml](https://github.com/tableau/connector-plugin-sdk/blob/master/connector-packager/connector_packager/package.py#L92), I think this change makes more sense to be called only once in `validate_all_xml`.

Reply:
Ok, yeah ideally it would fit within `validate_file_specific_rules`, but I think this is a good compromise for now.

Can you rename this method to `warn_file_specific_rules`

Reply:
Done, thanks